Penumbra/Api/PenumbraApi.cs

```diff
@@ -1014,6 +1014,20 @@ public Task ConvertTextureData(byte[] rgbaData, int width, string outputFile, Te
         };
     // @formatter:on
 
+    public IReadOnlyList<(string, string)> GetSerializedResourceTrees(ushort[] gameObjects)
+    {
+        var characters = gameObjects.Select(index => _dalamud.Objects[index]).OfType<Character>();
+        var resourceTrees = _resourceTreeFactory.FromCharacters(characters, ResourceTreeFactory.Flags.RedactExternalPaths |
+        ResourceTreeFactory.Flags.WithUiData |
```

Comment:
do we really want UIData here? I don't think so? Unsure about Ownership, too.

Reply:
Using UIData to draw the items associated with each texture. This is based off the same data reported in the On-Screen view.
I'll agree that ownership isn't needed though, I don't actually think it's currently being used when building the ResourceTree anyways

Reply:
Then maybe make the flag an input for the IPC too, because if you just want to export something you probably don't need to draw it, later on?

Reply:
The flags are scoped to Penumbra.Interop.ResourceTree so aren't accessible from the Api project

Reply:
and by UIData I mean that flag is used to populate the itemname fields on ResourceNodes, otherwise they're not set

Reply:
I think if this is not something that every consumer needs (and a consumer that just exports a model wouldn't), we should add a bool then to the IPC.
Changing or adding IPC after the fact is always ugly so I'd prefer to do this beforehand.

Reply:
could move the ResourceTreeFactory flags to the api project perhaps? then pass it as a parameter?

Reply:
I'd just give it a bool param `withUiData` and handle that with flags internally.